src/drivers/mct/main/prep_rof_mod.F90

```diff
@@ -354,8 +488,12 @@ subroutine prep_rof_merge(l2x_r, fractions_r, x2r_r, cime_model)
        if (have_irrig_field) then
           index_x2r_Flrl_irrig  = mct_aVect_indexRA(x2r_r,'Flrl_irrig' )
        end if
+       index_l2x_Flrl_Tqsur = mct_aVect_indexRA(l2x_r,'Flrl_Tqsur' )
+       index_l2x_Flrl_Tqsub = mct_aVect_indexRA(l2x_r,'Flrl_Tqsub' )
+       index_x2r_Flrl_Tqsur = mct_aVect_indexRA(x2r_r,'Flrl_Tqsur' )
+       index_x2r_Flrl_Tqsub = mct_aVect_indexRA(x2r_r,'Flrl_Tqsub' )	   
```

Comment:
@billsacks, sorry, this code is unfamiliar with me. I see three new blocks of index calls, these Flrl calls here, a bunch of a2x calls below, followed by a bunch of x2r calls. I assume we should be checking that some mct_aVect_indexRA of some field is not zero, but I don't know what field to use for these three blocks.

Reply:
Yes: checking here or in initialization if the index is 0, as is done for setting `have_irrig_field`:

```Fortran
       index_irrig = mct_aVect_indexRA(x2r_rx, irrig_flux_field, perrWith='quiet')
       if (index_irrig == 0) then
          have_irrig_field = .false.
       else
          have_irrig_field = .true.
       end if
```

or flds_wiso_rof:

```Fortran
       index_l2x_Flrl_rofl_16O = mct_aVect_indexRA(l2x_r,'Flrl_rofl_16O', perrWith='quiet' )

       if ( index_l2x_Flrl_rofl_16O /= 0 ) flds_wiso_rof = .true.
```

I'm also not sure exactly what checks you'll need here; it may depend somewhat on the logic that @rljacob adds in `seq_flds_mod`. Basically: according to our conversation last week, we were asking for these new fields to only be added conditionally. But for any field that is only added conditionally, I think there need to be conditionals here so that we don't get errors when trying to reference the index of an undefined field.

Reply:
@billsacks I can add new  `have_ <*>_field` similar to `have_irrig_field`. Does that sound good?
@jgfouca, How should I make the changes? In E3SM? Or, directly in your cime branch?

Reply:
I'm already working on this.